mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp

```diff
@@ -882,5 +900,79 @@ LogicalResult tuningTableLookup(TuningTable *perfTable, ModuleOp &mod,
   return failure();
 }
 
+template <typename ParamType>
+int64_t retrieveSplitKValueImpl(StringRef perfConfig) {
+  ParamType params;
+  params.deserialize(perfConfig.str());
+  return params.splitKFactor;
+}
+
+int64_t retrieveSplitKValue(rock::RockGemmWrapperInterface op,
+                            StringRef perfConfig) {
+  rock::GemmFeatures features = op.getGemmFeatures();
+  if (isAccel(features)) {
+    return retrieveSplitKValueImpl<rock::InitParamsAccel>(perfConfig);
+  }
+  return retrieveSplitKValueImpl<rock::InitParamsNonAccel>(perfConfig);
+}
+
+bool isSplitKRequested(ModuleOp &mod, StringRef &perfConfig) {
+  WalkResult gemmWalkResult =
+      mod.walk([&](rock::RockGemmWrapperInterface op) -> WalkResult {
+        int64_t splitKFactor = retrieveSplitKValue(op, perfConfig);
+        if (splitKFactor > 1) {
+          return WalkResult::interrupt();
+        }
+        return WalkResult::advance();
+      });
+
+  return gemmWalkResult.wasInterrupted();
+}
+
+RocmlirSplitKSelectionLikelihood isSplitKFaster(int64_t gDim, int64_t mDim,
+                                                int64_t nDim, int64_t kDim,
+                                                int64_t numCUs) {
+
+  // M/block N/block K/block M/wave N/wave kPack
+  const std::vector<std::vector<uint32_t>> rangeGemmParams = {
+      {4, 8, 16, 32, 64, 128, 256},
+      {16, 32, 64, 128, 256},
+      {1, 2, 4, 8},
+      {1, 4, 8, 16}};
+
+  rock::GemmSize gemmSize(gDim, mDim, kDim, nDim);
+  llvm::SmallSetVector<int64_t, 10> splitKValues = {};
+  double minWorkImbalance = std::numeric_limits<double>::max();
```

Comment:
We'll definitely want a comment making sure this is kept up-to-date with the tuning ranges.

Also, said tuning ranges are datatyp dependent.

Reply:
Put a comment with the reference where the loops were taken

Reply:
done